File: client/android/UseCase/Navigation/Shell/LoadNavigationShellUseCase.kt
package app.mobiling.client.usecase.navigation.shell

import app.mobiling.client.contract.navigation.shell.MobileNavigationShellPayload
import app.mobiling.client.data.navigation.shell.NavigationShellGateway

class LoadNavigationShellUseCase(
    private val gateway: NavigationShellGateway,
) {
    suspend operator fun invoke(): MobileNavigationShellPayload = gateway.loadMobileShell()
}

File: client/ios/UseCase/Navigation/Shell/LoadNavigationShellUseCase.swift
import Foundation

public struct LoadNavigationShellUseCase {
    private let gateway: NavigationShellGateway

    public init(gateway: NavigationShellGateway) {
        self.gateway = gateway
    }

    public func callAsFunction() async throws -> MobileNavigationShellPayload { try await gateway.loadMobileShell() }
}
